src/compass/utils/runconfig.py

```diff
@@ -246,65 +307,20 @@ def load_from_yaml(cls, yaml_path: str, workflow_name: str) -> RunConfig:
         workflow_name: str
             Name of the workflow for which uploading default options
         """
-        error_channel = journal.error('RunConfig.load_from_yaml')
-        try:
-            # Load schema corresponding to 'workflow_name' and to validate against
-            schema = yamale.make_schema(
-                f'{helpers.WORKFLOW_SCRIPTS_DIR}/schemas/cslc_s1.yaml',
-                parser='ruamel')
-        except:
-            err_str = f'unable to load schema for workflow {workflow_name}.'
-            error_channel.log(err_str)
-            raise ValueError(err_str)
-
-        # load yaml file or string from command line
-        if os.path.isfile(yaml_path):
-            try:
-                data = yamale.make_data(yaml_path, parser='ruamel')
-            except yamale.YamaleError as yamale_err:
-                err_str = f'Yamale unable to load {workflow_name} runconfig yaml {yaml_path} for validation.'
-                error_channel.log(err_str)
-                raise yamale.YamaleError(err_str) from yamale_err
-        else:
-            raise FileNotFoundError
-
-        # validate yaml file taken from command line
-        try:
-            yamale.validate(schema, data)
-        except yamale.YamaleError as yamale_err:
-            err_str = f'Validation fail for {workflow_name} runconfig yaml {yaml_path}.'
-            error_channel.log(err_str)
-            raise yamale.YamaleError(err_str) from yamale_err
-
-        # load default runconfig
-        parser = YAML(typ='safe')
-        default_cfg_path = f'{helpers.WORKFLOW_SCRIPTS_DIR}/defaults/cslc_s1.yaml'
-        with open(default_cfg_path, 'r') as f_default:
-            default_cfg = parser.load(f_default)
-
-        with open(yaml_path, 'r') as f_yaml:
-            user_cfg = parser.load(f_yaml)
-
-        # Copy user-supplied configuration options into default runconfig
-        helpers.deep_update(default_cfg, user_cfg)
-
-        # Validate YAML values under groups dict
-        validate_group_dict(default_cfg['runconfig']['groups'])
+        cfg = load_validate_yaml(yaml_path, workflow_name)
 
         # Convert runconfig dict to SimpleNamespace
-        sns = wrap_namespace(default_cfg['runconfig']['groups'])
+        sns = wrap_namespace(cfg['runconfig']['groups'])
 
         bursts = runconfig_to_bursts(sns)
 
         # Load reference grids if not reference run i.e. not running rdr2geo
```

Comment:
```suggestion
        # Load reference grids if not reference run i.e. not running rdr2geo
        ref_rdr_grid_info = None
```

This seems to fix the UnboundLocalError of `ref_rdr_grid_info` while testing the reference rslc